src/SfxWeb/src/app/views/application-type/create-application/create-application.component.ts

```diff
@@ -12,29 +14,33 @@ import { Constants } from 'src/app/Common/Constants';
 })
 export class CreateApplicationComponent implements OnInit {
 
-  app: ApplicationType;
+  @Input() inputs: { appType: ApplicationType };
   form: UntypedFormGroup;
+  disableSubmit = new EventEmitter<boolean>();
 
   constructor(public dialogRef: MatDialogRef<CreateApplicationComponent>,
               @Inject(MAT_DIALOG_DATA) public data: IsolatedAction,
               private formBuilder: UntypedFormBuilder) { }
 
   ngOnInit(): void {
-   this.app = this.data.data.appType;
-   this.form = this.formBuilder.group({
-    userInput: [Constants.FabricPrefix + this.app.name, [Validators.required, Validators.pattern(/^fabric:\/.+/)]]
+    this.form = this.formBuilder.group({
+      userInput: [Constants.FabricPrefix + this.inputs.appType.name, [Validators.required, Validators.pattern(/^fabric:\/.+/)]]
     });
+    this.form.valueChanges.subscribe(() => {
```

Comment:
this should be unsubscribed from in an OnDestroy otherwise we leak subscriptions

Reply:
Added OnDestroy to the two instances of form validity checker. What about the subscribe inside action-dialog.ts and utils.ts? Are those safe?

Reply:
it depends if the observable is closed in those.